src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageAccountProvider.cs

```diff
@@ -15,6 +16,7 @@ internal class DefaultStorageAccountProvider : IStorageAccountProvider
         private readonly IStorageCredentialsValidator _storageCredentialsValidator;
         private readonly IStorageAccountParser _storageAccountParser;
         private readonly IServiceProvider _services;
+        private readonly ConcurrentDictionary<string, Task<IStorageAccount>> _accounts = new ConcurrentDictionary<string, Task<IStorageAccount>>();
```

Comment:
I think caching the Task is incorrect - we want a **value** cache. That's what makes sense conceptually.